consensus/consensus_tests/utils_test.go

```diff
@@ -306,17 +306,22 @@ func basePersistenceMock(t *testing.T, _ modules.EventsChannel) *modulesMock.Moc
 	ctrl := gomock.NewController(t)
 	persistenceMock := modulesMock.NewMockPersistenceModule(ctrl)
 	persistenceContextMock := modulesMock.NewMockPersistenceRWContext(ctrl)
+	persistenceReadContextMock := modulesMock.NewMockPersistenceReadContext(ctrl)
 
 	persistenceMock.EXPECT().Start().Return(nil).AnyTimes()
 	persistenceMock.EXPECT().SetBus(gomock.Any()).Return().AnyTimes()
-	persistenceMock.EXPECT().NewReadContext(int64(-1)).Return(persistenceContextMock, nil).AnyTimes()
+	persistenceMock.EXPECT().NewReadContext(int64(-1)).Return(persistenceReadContextMock, nil).AnyTimes()
 	persistenceMock.EXPECT().ReleaseWriteContext().Return(nil).AnyTimes()
 
 	// The persistence context should usually be accessed via the utility module within the context
 	// of the consensus module. This one is only used when loading the initial consensus module
 	// state; hence the `-1` expectation in the call above.
 	persistenceContextMock.EXPECT().Close().Return(nil).AnyTimes()
-	persistenceContextMock.EXPECT().GetLatestBlockHeight().Return(uint64(0), nil).AnyTimes()
+	persistenceReadContextMock.EXPECT().GetLatestBlockHeight().Return(uint64(0), nil).AnyTimes()
+
+	// here we are mocking 4 actors since we are considering the 4 nodes in the LocalNet genesis
```

Comment:
I know it might be overkill, but if we use a repeat function (not available for structs by default), you can reuse the `numValidators` parameter: https://go.dev/play/p/uWDXFeh5d_o because that's what runTimeMgr is dependant on

Reply:
Yeah, me likey generics but I'd rather go with something like this https://github.com/pokt-network/pocket/pull/378/commits/7c051965290273863892406755f36914d7c83132